<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Lars Hagstr√∂m</author>
 <class>MemoryGraph</class>
 <widget class="QWidget" name="MemoryGraph">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>518</width>
    <height>379</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>10000</width>
    <height>10000</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Dob shared memory usage</string>
  </property>
  <property name="toolTip">
   <string>Memory information and statistics.</string>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="1,98,0">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="memLabel">
       <property name="text">
        <string>mem</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>History</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="historyLength">
       <property name="toolTip">
        <string>Change the horizontal scale of the graph.</string>
       </property>
       <property name="prefix">
        <string/>
       </property>
       <property name="suffix">
        <string> min</string>
       </property>
       <property name="decimals">
        <number>0</number>
       </property>
       <property name="minimum">
        <double>1.000000000000000</double>
       </property>
       <property name="maximum">
        <double>172800.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>10.000000000000000</double>
       </property>
       <property name="value">
        <double>20.000000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Update Period</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="updatePeriod">
       <property name="toolTip">
        <string>Change how often the graph is updated.</string>
       </property>
       <property name="suffix">
        <string> s</string>
       </property>
       <property name="decimals">
        <number>0</number>
       </property>
       <property name="minimum">
        <double>1.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10000.000000000000000</double>
       </property>
       <property name="value">
        <double>3.000000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="GraphWidget" name="graph">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>100</width>
         <height>100</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>10000</width>
         <height>10000</height>
        </size>
       </property>
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This graph shows the amount of memory allocated in the Dob shared memory. &lt;/p&gt;&lt;p&gt;The total amount of available memory is configured in Safir.Dob.NodeParameters.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Sunken</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="scalePercent">
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>100 %</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="verticalScale">
         <property name="toolTip">
          <string>Set the vertical scale of the memory usage graph.</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>  0 %</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,4,1,2,1,4,1,2,2,1">
     <item>
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Total:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="total">
       <property name="toolTip">
        <string>Total memory allocated for Dob shared memory.</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="maxLength">
        <number>5</number>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>MiB</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_4">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Current</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="current">
       <property name="toolTip">
        <string>Currently allocated memory of the Dob shared memory.</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="maxLength">
        <number>5</number>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>MiB</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLineEdit" name="currentPercent">
       <property name="toolTip">
        <string>Percent of the Dob shared memory that is allocated.</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="maxLength">
        <number>3</number>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>%</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GraphWidget</class>
   <extends>QFrame</extends>
   <header>graphwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
